### PR TITLE
Subscriptions Form button fix and tests  

### LINKED ISSUE #6931: T - No way to subscribe after canceling Stripe modal
**Summary**
On the [payments landing page](https://developer.allizom.org/en-US/payments/), after clicking "continue" the Stripe modal comes up. After canceling the stripe modal means, the "continue" button becomes inactive. Only reloading the page turns it active again.

**Steps To Reproduce (STR)**
1. Open https://developer.allizom.org/en-US/payments/
2. Click the checkbox and then "continue"
3. Cancel the Stripe modal by clicking on the x
4. Click on the "continue" button

**Actual behavior**
"continue" button is inactive in step 4

**Expected behavior**
"continue" button should bring up the Stripe modal again

### PR DESCRIPTION
**Pull request by mindy**
Changes:
- Fix bug in #6931, where Stripe modal couldn't open a second time. 
- Add some tests. Initially I wanted to cover the full component but it was taking up too much time (I didn't want to spend all day on writing a test!), so I committed what I had so far. 
- Reduce size of "Sorry!" heading
- Consolidate payments error messages 

To test: 
- Go to /payments and ensure making a subscription still works. 
- Ensure error messages still work and look ok: 
sorry-text

Closes #6931

### CODE REVIEW

File: kuma/javascript/src/payments/components/subscription-form.jsx [hunk 1]
```diff
@@ -9,45 +9,11 @@ import GAProvider, {
 } from '../../ga-provider.jsx';
 import UserProvider from '../../user-provider.jsx';
 import { getCookie } from '../../utils';
+import { ErrorComponent } from './errors.jsx';
+import useScriptLoading from './useScriptLoading.js';
```
Comment (mindy) [inline kuma/javascript/src/payments/components/subscription-form.jsx:13]: In order to mock this function, I had to move it to its own file. I tried other ways around this but couldn't get it to work.

File: kuma/javascript/src/payments/components/subscription-form.jsx [hunk 2]
```diff
@@ -247,47 +214,31 @@ export default function SubscriptionForm() {
                 // If window.mdn.triggerAuthModal is falsy, it most likely means
                 // it deliberately doesn't want this user to use a modal. E.g.
                 // certain mobile clients.
-                window.location.href = `/${locale}/users/account/signup-landing?next=${next}`;
+                window.location.assign(
```
Comment (mindy) [inline kuma/javascript/src/payments/components/subscription-form.jsx:217]: Changing it to `.assign()` allows us to test it.

File: kuma/javascript/src/payments/components/subscription-form.jsx [hunk 3]
```diff
@@ -247,47 +214,31 @@ export default function SubscriptionForm() {
...
+                    `/${locale}/users/account/signup-landing?next=${next}`
+                );
             }
         }
     }
 
     let content;
     if (formStep === 'server_error') {
         content = (
-            <section className="error">
```
Comment (mindy) [inline kuma/javascript/src/payments/components/subscription-form.jsx:258]: Noticed this and the error message below were similar so I thought I'd extract them into one component.

File: kuma/javascript/src/payments/components/subscription-form.jsx [hunk 4]
```diff
@@ -199,6 +165,7 @@ export default function SubscriptionForm() {
                         createSubscription();
                     },
                     closed() {
+                        setOpenStripeModal(false);
```
Comment (mindy) [inline kuma/javascript/src/payments/components/subscription-form.jsx:168]: This is the line that fixes the ticket (https://github.com/mdn/kuma/issues/6931)

File: kuma/javascript/src/payments/components/errors.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+// @flow
+import * as React from 'react';
+import { gettext, Interpolated } from '../../l10n.js';
+
+type Props = {
+    text: string,
+    onClick?: (event: SyntheticEvent<HTMLAnchorElement>) => void,
```
Comment (Gregoor) [inline kuma/javascript/src/payments/components/errors.jsx:7]: Looks like all the instances use the retry button atm, so I'd go with making `onClick` a required prop

File: kuma/javascript/src/payments/components/useScriptLoading.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+// @flow
```
Comment (Gregoor) [inline kuma/javascript/src/payments/components/useScriptLoading.js:1]: file naming nit: this file is camelCase whereas we mostly use snake case all over the frontend files.

File: kuma/javascript/src/payments/components/useScriptLoading.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import { useEffect, useState } from 'react';
+
+/**
+ * Loads the script given by the URL and cleans up after itself
+ * @param string url of the script to load
```
Comment (Gregoor) [inline kuma/javascript/src/payments/components/useScriptLoading.js:6]: `string` should be `{string}`
https://jsdoc.app/tags-param.html

File: kuma/javascript/src/payments/components/useScriptLoading.js (hunk 6 above)
Comment (peterbe) [reply kuma/javascript/src/payments/components/useScriptLoading.js:1]: I agree. But it's actually not "snake case" all over. It's "lower train case". Yeah, I made that up. E.g. `cancel-subscription-form.jsx` or `feedback-form.jsx`.

File: kuma/javascript/src/payments/components/useScriptLoading.js (hunk 7 above)
Comment (peterbe) [reply kuma/javascript/src/payments/components/useScriptLoading.js:6]: Gawd I'm not going to miss these sporadic jsdoc comments when we move the Yari front-end to TS. BUT, I often like the doc string comment that sometimes explains why the function exists.

File: kuma/javascript/src/payments/components/errors.jsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+};
+
+/**
+ * Structured error message with header, text, and optional button
+ */
+const ErrorComponent = ({ text, onClick }: Props): React.Node => (
```
[the remark this answers is not in the document]
Comment (mindy) [reply kuma/javascript/src/payments/components/errors.jsx:13]: Changed to `ErrorWithRetry`

**Comment by mindy** [comment]
Ready for another 👀 @Gregoor. Thank you!

**Pull request merged**